server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/utility/CICSSendOptionsCheckUtility.java

```diff
@@ -0,0 +1,250 @@
+          put(CICSLexer.MAPSET, ErrorSeverity.ERROR);
+          put(CICSLexer.DATAONLY, ErrorSeverity.WARNING);
+          put(CICSLexer.MAPONLY, ErrorSeverity.WARNING);
+          put(CICSLexer.NLEOM, ErrorSeverity.WARNING);
+          put(CICSLexer.FMHPARM, ErrorSeverity.ERROR);
+          put(CICSLexer.NOFLUSH, ErrorSeverity.WARNING);
+          put(CICSLexer.MAPPINGDEV, ErrorSeverity.ERROR);
+          put(CICSLexer.PAGE, ErrorSeverity.ERROR);
+          put(CICSLexer.RELEASE, ErrorSeverity.WARNING);
+          put(CICSLexer.TRANSID, ErrorSeverity.ERROR);
+          put(CICSLexer.TRAILER, ErrorSeverity.ERROR);
+          put(CICSLexer.RETAIN, ErrorSeverity.WARNING);
+          put(CICSLexer.AUTOPAGE, ErrorSeverity.WARNING);
+          put(CICSLexer.CURRENT, ErrorSeverity.WARNING);
+          put(CICSLexer.ALL, ErrorSeverity.WARNING);
+          put(CICSLexer.NOAUTOPAGE, ErrorSeverity.WARNING);
+          put(CICSLexer.OPERPURGE, ErrorSeverity.WARNING);
+          put(CICSLexer.PARTNSET, ErrorSeverity.ERROR);
+          put(CICSLexer.TEXT, ErrorSeverity.ERROR);
+          put(CICSLexer.HEADER, ErrorSeverity.ERROR);
+          put(CICSLexer.JUSTIFY, ErrorSeverity.ERROR);
+          put(CICSLexer.JUSFIRST, ErrorSeverity.WARNING);
+          put(CICSLexer.JUSLAST, ErrorSeverity.WARNING);
+          put(CICSLexer.MAPPED, ErrorSeverity.WARNING);
+          put(CICSLexer.NOEDIT, ErrorSeverity.WARNING);
+        }
+      };
+
+
+  public CICSSendOptionsCheckUtility(DialectProcessingContext context, List<SyntaxError> errors) {
+    super(context, errors, DUPLICATE_CHECK_OPTIONS);
+  }
+
+  /**
+   * Entrypoint to check CICS Send rule options
+   *
+   * @param ctx ParserRuleContext subclass containing options
+   * @param <E> A subclass of ParserRuleContext
+   */
+  public <E extends ParserRuleContext> void checkOptions(E ctx) {
+    switch (ctx.getRuleIndex()) {
+      case CICSParser.RULE_cics_send_group1:
+        checkGroup1((CICSParser.Cics_send_group1Context) ctx);
+        break;
+      case CICSParser.RULE_cics_send_control_map:
+        checkControl((CICSParser.Cics_send_control_mapContext) ctx);
+        break;
+      case CICSParser.RULE_cics_send_mappingdev:
+        checkMappingdev((CICSParser.Cics_send_mappingdevContext) ctx);
+        break;
+      case CICSParser.RULE_cics_send_page:
+        checkPage((CICSParser.Cics_send_pageContext) ctx);
+        break;
+      case CICSParser.RULE_cics_send_text:
+        checkText((CICSParser.Cics_send_textContext) ctx);
+        break;
+      case CICSParser.RULE_cics_send_text_mapped:
+        checkTextMapped((CICSParser.Cics_send_text_mappedContext) ctx);
+        break;
+      case CICSParser.RULE_cics_send_text_noedit:
+        checkTextNoedit((CICSParser.Cics_send_text_noeditContext) ctx);
+        break;
+      default:
+        break;
+    }
+    checkDuplicates(ctx);
+  }
+
+  @SuppressWarnings("unchecked")
+  private void checkGroup1(CICSParser.Cics_send_group1Context ctx) {
```

Comment:
I think better to add missing message for LENGTH or FLENGTH requires FROM in all groups